network-api/networkapi/wagtailpages/pagemodels/buyersguide/consumer_creepometer_page.py

```diff
@@ -28,3 +30,13 @@ class ConsumerCreepometerPage(BasePage):
     translatable_fields = [
         SynchronizedField("year"),
     ]
+
+    @property
+    def bg_page(self):
+        active_locale = Locale.get_active()
+        return BuyersGuidePage.objects.get(locale=active_locale)
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        context["methodology_url"] = self.bg_page.url + self.bg_page.reverse_subpage("methodology-view")
+        return context
```

Comment:
I have a bit more context now with the PR :smiley: I think we can improve this a bit more.

It turns out that the `ConsumerCreepometerPage` has to be a child of `BuyersGuideEditoralContextIndexPage` which is itself a child of the `BuyersGuidePage`. Hence, this page must be a grandchild of the `BuyersGuidePage` and we can simplify that to:

````
    def get_context(self, request, *args, **kwargs):
        context = super().get_context(request, *args, **kwargs)
        bg_page = self.get_parent().get_parent().specific()
        context["methodology_url"] = bg_page.url + bg_page.reverse_subpage("methodology-view")
        return context
````
(I think that should work :crossed_fingers: )

Reply:
🤯 
Thanks @jhonatan-lopes , let me update this PR accordingly!